File: Output/python/easy/SingleNumber.py
#https://leetcode.com/problems/single-number/
#Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
#You must implement a solution with a linear runtime complexity and use only constant extra space.


class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res=0
        for i in range(len(nums)):
            res=res^nums[i]
        return res

    

File: Tool/PreUtil/src/output/javascript/easy/SingleNumber.js
/**https://leetcode.com/problems/single-number/ */
//Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
//You must implement a solution with a linear runtime complexity and use only constant extra space.


/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let result = 0;
    for(let i = 0; i < nums.length; i++){
        result ^= nums[i];
    }
    return result;
    
};

    